packages/react-compose/etc/react-compose.api.md

```diff
@@ -10,64 +10,70 @@ import * as React from 'react';
 export type ClassDictionary = Record<string, string>;
 
 // @public (undocumented)
-export interface ComponentWithAs<E extends React.ElementType = 'div', P = {}> extends React.FunctionComponent {
+export interface ComponentWithAs<TElementType extends React.ElementType = 'div', TProps = {}> extends React.FunctionComponent {
     // (undocumented)
-    <EE extends React.ElementType = E>(props: Omit<PropsOfElement<EE>, 'as' | keyof P> & {
-        as?: EE;
-    } & P): JSX.Element | null;
+    <TExtendedElementType extends React.ElementType = TElementType>(props: Omit<PropsOfElement<TExtendedElementType>, 'as' | keyof TProps> & {
+        as?: TExtendedElementType;
+    } & TProps): JSX.Element | null;
     // (undocumented)
-    defaultProps?: Partial<P & {
-        as: E;
+    defaultProps?: Partial<TProps & {
+        as: TElementType;
     }>;
     // (undocumented)
     displayName?: string;
     // (undocumented)
-    propTypes?: React.WeakValidationMap<P> & {
+    propTypes?: React.WeakValidationMap<TProps> & {
         as: React.Requireable<string | ((props: any, context?: any) => any) | (new (props: any, context?: any) => any)>;
     };
 }
 
-// @public (undocumented)
-export function compose<ElementType extends React.ElementType, InputProps, InputStylesProps, ParentProps, ParentStylesProps>(input: Input<ElementType, InputProps>, inputOptions?: ComposeOptions<InputProps, InputStylesProps, ParentProps, ParentStylesProps>): ComponentWithAs<ElementType, InputProps & ParentProps>;
+// @public
+export function compose<TElementType extends React.ElementType, TProps = {}, TInputStylesProps = {}, TParentProps = {}, TParentStylesProps = {}, TState = TProps>(input: Input<TElementType, TProps>, inputOptions?: ComposeOptions<TProps, TInputStylesProps, TParentProps, TParentStylesProps>): ComponentWithAs<TElementType, TProps & TParentProps>;
 
 // @public (undocumented)
-export type ComposedComponent<P = {}> = React.FunctionComponent<P> & {
-    fluentComposeConfig: ComposePreparedOptions;
+export type ComposedComponent<TElementType extends React.ElementType<any> = 'div', TProps = {}, TState = TProps> = React.FunctionComponent<TProps> & {
+    fluentComposeConfig: ComposePreparedOptions<TElementType, TProps, TState>;
 };
 
 // @public (undocumented)
-export type ComposeOptions<InputProps = {}, InputStylesProps = {}, ParentProps = {}, ParentStylesProps = {}> = {
+export type ComposeOptions<TInputProps = {}, TInputStylesProps = {}, TParentProps = {}, TParentStylesProps = {}> = {

```

Comment:
Would be better to stage the generics rename as a separate PR, just for easier reviewing :)